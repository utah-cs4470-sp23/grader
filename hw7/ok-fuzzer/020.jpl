let a = false
let b[c] = [false, (false == false)]
fn d() : bool {
  let {{{}, e[f, g, h], i}, {j[k], {l, m}}, n} = (array[e : (- (c % argnum)), f : 666] {{{}, (array[g : argnum, h : argnum, i : 246] g), (f * argnum)}, {b, {a, f}}, (array[g : c, h : (if false then e else 611), i : (sum[g : e] f)] atan2(80.63, 57.51))})[(array[e : (sum[e : (sum[e : argnum, f : c, g : argnum] g), f : (sum[e : c] argnum)] 807), f : ((sum[e : argnum] e) + c)] f)[(c % (390 / (sum[e : 106] e))), ((sum[e : (sum[e : 495] argnum), f : (- c), g : (sum[e : c, f : c, g : argnum] e)] (- f)) / (c + (c - c)))], {(- c), (! false), ((945 < argnum) && (! a))}{0}]
  let {} = (if ((exp(to_float(h)) == sin(exp(79.35))) || l) then {} else {{}}{0})
  let o[p, q, r] = e
  assert [d(), false][i], "s"
  return false
}

fn e({ f[g, h, i] : bool[,,], j[k, l, m] : bool[,,] }, n[o, p] : {float}[,], q : int) : bool {
  let r[s, t] = (if (! (false && (! (false || a)))) then (if (! ((- k) == h)) then (if (! d()) then n else (array[r : l] n)[(- c)]) else (if (l < i) then (array[r : i, s : 353, t : p] n)[118, argnum, (if a then 732 else o)] else n)) else {n}{0})
  return (if (to_int(sqrt(log(60.63))) <= l) then (if a then (if false then {n, false} else {n, a}){1} else true) else a)
  let u = i
  let {v, {}, {w, x[y, z, A], B}} = {{p, (array[v : m, w : h, x : u] {})[(- 200), (if d() then k else c), i], (if (i != argnum) then {h, j, r} else {i, j, r})}, q}{0}
  return (asin(pow(to_float(l), (sum[C : A, D : w] sqrt(24.81)))) > sin(tan(acos(to_float(z)))))
}

let {f} = (if (if false then {b[argnum], (false || d())} else (if (d() || a) then {a, false} else {a, true})){1} then {(if (argnum != 480) then {17.49} else {73.35})}{0} else (array[f : 744, g : c] {to_float(f)})[813, (- (if d() then argnum else (- c)))])
write (array[g : c, h : (if d() then (120 + 148) else to_int(cos(47.61)))] (array[i : (array[i : g, j : g] g)[h, (if true then c else g)], j : h] {(- f), asin(f), cos(75.01), log(f)}))[(188 * argnum), (sum[g : c, h : argnum] to_int(atan2((66.87 / 72.15), asin(f))))] to "g.png"

let g[h, i] = {{(array[g : argnum, h : c] 77.89)}{0}}{0}
fn j(k[l] : float[]) : bool {
  let m[n, o, p] = (if j((if (! true) then {false, l, k} else {d(), 340, k}){2}) then (array[m : ((array[m : 372, n : h] 262)[(c % l), l] / i), n : to_int(acos((sum[m : 550, n : l] f))), o : argnum] false) else (array[m : (if a then argnum else argnum), n : argnum] (array[o : i, p : (88 / n), q : l] a))[(if ((- 55.91) >= sqrt(36.94)) then i else {g, 349}{1}), (sum[m : i, n : i, o : (h - i)] i)])
  let {q[r, s, t], {u}, v[w, x]} = (array[q : (if a then (array[q : argnum, r : n] argnum) else (array[q : p, r : argnum] argnum))[(array[q : p, r : n, s : 256] c)[(i - i), argnum, (if a then i else h)], (- (sum[q : l, r : h] p))]] {(array[r : h] m)[(- c)], {(array[r : p, s : h, t : n] 5.66)}, g})[{c, tan(atan2(f, f)), 432}{0}]
  let {y[z, A, B]} = {argnum, g, {85.91, to_float(w), {u}}}{2}{2}
  let C[D, E, F] = {(if j(k) then {y, args}{0} else y)}{0}
  return ((o * o) <= to_int(log(to_float(E))))
}

let k = (if {d(), {}}{0} then (if (i != (sum[k : c, l : argnum, m : 457] i)) then {true, to_int(4.12)} else {(! a), 865}) else (if a then {(if d() then d() else a), argnum} else {(false || a), (sum[k : 570] k)})){1}
time time let l[m] = b


fn n() : bool {
  let {o[p, q], {r, s[t, u]}, {}} = {(array[o : h, p : {c}{0}] g)[c, (if (k != (- k)) then (if (k != h) then argnum else argnum) else (- 715))], {{g[h, k], (array[o : m, p : 970] 358)}, j((array[o : 42] 26.06))}{0}, {}}
  assert (d() || e((if (! (d() == true)) then {(array[v : p, w : m, x : p] true), (array[v : p, w : q, x : u] a)} else {(array[v : h, w : c, x : argnum] d()), (array[v : c, w : t, x : i] d())}), (array[v : (- q), w : p] {sin(r)}), q)), "v"
  let v = (! e({(array[v : 901, w : p, x : t] n()), (array[v : i, w : k, x : argnum] (c <= p))}, (if ((! true) == a) then (array[v : h, w : (argnum + u)] {r}) else (array[v : c, w : i] {f})), h))
  let w[x, y, z] = (if (q > (if (n() && d()) then (array[w : 612, x : c, y : argnum] p) else (array[w : u, x : 634, y : t] t))[k, h, k]) then (if (b[m] || false) then {(array[w : m, x : c, y : h] x), r, o} else {(array[w : q, x : k, y : m] c), r, o}){0} else (if n() then (array[w : (i % q), x : (sum[w : 731, x : p, y : c] 231)] (array[y : 210, z : m, A : w] t))[u, q] else (if n() then (if true then (array[w : argnum, x : q, y : argnum] w) else (array[w : u, x : 10, y : t] x)) else (if a then (array[w : c, x : h, y : p] q) else (array[w : argnum, x : u, y : argnum] x)))))
  return v
}

fn o(p : bool, q : bool) : int[,,] {
  let r[s, t, u] = (array[r : h, s : to_int(log(83.56)), t : h] p)
  return {h, (array[v : s, w : h, x : t] (array[y : t, z : v, A : w] t))}{1}[(if a then {(a || d()), (k - s), r[u, s, m]} else {(if p then d() else a), to_int(f), false}){1}, {(array[v : k, w : m] i)}{0}[i, k], (if (814 >= u) then {(sum[v : c, w : s] w)} else (if n() then {u} else {m})){0}]
  return (array[v : {{h, a}, f}{0}{0}, w : m, x : (if (if (if p then q else d()) then true else (if d() then n() else a)) then o(q, (false || true)) else (if (! p) then o(q, a) else o(a, true)))[m, ({q, k, argnum}{1} * k), (if (! (! q)) then u else t)]] s)
  let {} = [(if p then [{}, {}, {}] else (array[v : u] {}))[{true, q, i}{2}]][k]
  return (array[v : ({(m % t), s}{0} + u), w : k, x : i] (to_int(pow(asin(84.52), f)) + (sum[y : i] h)))
}

