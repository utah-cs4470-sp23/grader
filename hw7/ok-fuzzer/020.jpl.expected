(AssertCmd
 (IfExpr
  (BoolType)
  (UnopExpr (BoolType) ! (FalseExpr (BoolType)))
  (ArrayIndexExpr
   (BoolType)
   (TupleIndexExpr
    (ArrayType (BoolType) 2)
    (TupleLiteralExpr
     (TupleType (ArrayType (BoolType) 2))
     (ArrayLoopExpr
      (ArrayType (BoolType) 2)
      a
      (IntExpr (IntType) 708)
      b
      (IntExpr (IntType) 914)
      (FalseExpr (BoolType))))
    0)
   (SumLoopExpr
    (IntType)
    a
    (ArrayIndexExpr
     (IntType)
     (ArrayLoopExpr
      (ArrayType (IntType) 1)
      a
      (IntExpr (IntType) 769)
      (IntExpr (IntType) 685))
     (BinopExpr (IntType) (IntExpr (IntType) 416) - (VarExpr (IntType) argnum)))
    b
    (SumLoopExpr
     (IntType)
     a
     (SumLoopExpr
      (IntType)
      a
      (IntExpr (IntType) 20)
      (VarExpr (IntType) argnum))
     (UnopExpr (IntType) - (VarExpr (IntType) a)))
    c
    (IntExpr (IntType) 233)
    (BinopExpr (IntType) (VarExpr (IntType) c) % (VarExpr (IntType) a)))
   (ArrayIndexExpr
    (IntType)
    (IfExpr
     (ArrayType (IntType) 3)
     (BinopExpr (BoolType) (FalseExpr (BoolType)) && (TrueExpr (BoolType)))
     (ArrayLoopExpr
      (ArrayType (IntType) 3)
      a
      (IntExpr (IntType) 964)
      b
      (IntExpr (IntType) 220)
      c
      (IntExpr (IntType) 128)
      (IntExpr (IntType) 434))
     (ArrayLoopExpr
      (ArrayType (IntType) 3)
      a
      (VarExpr (IntType) argnum)
      b
      (IntExpr (IntType) 817)
      c
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) c)))
    (VarExpr (IntType) argnum)
    (BinopExpr
     (IntType)
     (IfExpr
      (IntType)
      (TrueExpr (BoolType))
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) argnum))
     /
     (VarExpr (IntType) argnum))
    (ArrayIndexExpr
     (IntType)
     (ArrayLoopExpr
      (ArrayType (IntType) 2)
      a
      (VarExpr (IntType) argnum)
      b
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) argnum))
     (UnopExpr (IntType) - (IntExpr (IntType) 585))
     (SumLoopExpr
      (IntType)
      a
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) argnum)))))
  (TupleIndexExpr
   (BoolType)
   (TupleLiteralExpr (TupleType (BoolType)) (FalseExpr (BoolType)))
   0))
 "a")
(LetCmd (ArgLValue (VarArgument a)) (VarExpr (IntType) argnum))
(LetCmd
 (ArgLValue (VarArgument b))
 (BinopExpr
  (BoolType)
  (BinopExpr
   (BoolType)
   (VarExpr (IntType) argnum)
   >=
   (IfExpr
    (IntType)
    (BinopExpr
     (BoolType)
     (FloatExpr (FloatType) 0)
     >=
     (UnopExpr (FloatType) - (FloatExpr (FloatType) 31)))
    (UnopExpr
     (IntType)
     -
     (BinopExpr (IntType) (IntExpr (IntType) 924) - (IntExpr (IntType) 938)))
    (VarExpr (IntType) argnum)))
  &&
  (ArrayIndexExpr
   (BoolType)
   (ArrayLoopExpr
    (ArrayType (BoolType) 3)
    b
    (BinopExpr
     (IntType)
     (IfExpr
      (IntType)
      (TrueExpr (BoolType))
      (VarExpr (IntType) a)
      (IntExpr (IntType) 474))
     /
     (SumLoopExpr
      (IntType)
      b
      (VarExpr (IntType) a)
      c
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) b)))
    c
    (BinopExpr (IntType) (IntExpr (IntType) 756) * (VarExpr (IntType) a))
    d
    (VarExpr (IntType) argnum)
    (BinopExpr
     (BoolType)
     (BinopExpr (BoolType) (VarExpr (IntType) a) == (VarExpr (IntType) c))
     !=
     (FalseExpr (BoolType))))
   (VarExpr (IntType) a)
   (SumLoopExpr
    (IntType)
    b
    (IntExpr (IntType) 91)
    c
    (VarExpr (IntType) a)
    (BinopExpr (IntType) (VarExpr (IntType) a) + (VarExpr (IntType) argnum)))
   (VarExpr (IntType) argnum))))
(ShowCmd
 (ArrayLiteralExpr
  (ArrayType (TupleType (FloatType) (BoolType) (BoolType)) 1)
  (TupleLiteralExpr
   (TupleType (FloatType) (BoolType) (BoolType))
   (BinopExpr
    (FloatType)
    (UnopExpr
     (FloatType)
     -
     (BinopExpr
      (FloatType)
      (FloatExpr (FloatType) 54)
      -
      (FloatExpr (FloatType) 54)))
    /
    (UnopExpr (FloatType) - (FloatExpr (FloatType) 0)))
   (BinopExpr
    (BoolType)
    (IfExpr
     (IntType)
     (UnopExpr (BoolType) ! (VarExpr (BoolType) b))
     (VarExpr (IntType) argnum)
     (IntExpr (IntType) 710))
    !=
    (VarExpr (IntType) a))
   (BinopExpr
    (BoolType)
    (ArrayIndexExpr
     (FloatType)
     (ArrayLoopExpr
      (ArrayType (FloatType) 3)
      c
      (VarExpr (IntType) argnum)
      d
      (VarExpr (IntType) a)
      e
      (VarExpr (IntType) a)
      (FloatExpr (FloatType) 64))
     (VarExpr (IntType) a)
     (IntExpr (IntType) 562)
     (IfExpr
      (IntType)
      (FalseExpr (BoolType))
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) a)))
    >
    (ArrayIndexExpr
     (FloatType)
     (ArrayLiteralExpr
      (ArrayType (FloatType) 1)
      (FloatExpr (FloatType) 35)
      (FloatExpr (FloatType) 80))
     (SumLoopExpr
      (IntType)
      c
      (VarExpr (IntType) argnum)
      d
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) d)))))
  (TupleLiteralExpr
   (TupleType (FloatType) (BoolType) (BoolType))
   (FloatExpr (FloatType) 75)
   (VarExpr (BoolType) b)
   (BinopExpr
    (BoolType)
    (FalseExpr (BoolType))
    ||
    (IfExpr
     (BoolType)
     (BinopExpr (BoolType) (TrueExpr (BoolType)) || (VarExpr (BoolType) b))
     (BinopExpr
      (BoolType)
      (FloatExpr (FloatType) 93)
      !=
      (FloatExpr (FloatType) 85))
     (FalseExpr (BoolType)))))
  (TupleLiteralExpr
   (TupleType (FloatType) (BoolType) (BoolType))
   (SumLoopExpr
    (FloatType)
    c
    (BinopExpr (IntType) (IntExpr (IntType) 790) - (VarExpr (IntType) a))
    (SumLoopExpr
     (FloatType)
     d
     (ArrayIndexExpr
      (IntType)
      (VarExpr (ArrayType (IntType) 1) args)
      (VarExpr (IntType) c))
     e
     (BinopExpr (IntType) (IntExpr (IntType) 373) + (IntExpr (IntType) 531))
     f
     (IfExpr
      (IntType)
      (VarExpr (BoolType) b)
      (IntExpr (IntType) 895)
      (IntExpr (IntType) 459))
     (UnopExpr (FloatType) - (FloatExpr (FloatType) 59))))
   (TupleIndexExpr
    (BoolType)
    (TupleLiteralExpr
     (TupleType (BoolType) (BoolType))
     (VarExpr (BoolType) b)
     (BinopExpr (BoolType) (TrueExpr (BoolType)) || (VarExpr (BoolType) b)))
    1)
   (BinopExpr
    (BoolType)
    (BinopExpr (BoolType) (VarExpr (IntType) argnum) == (VarExpr (IntType) a))
    &&
    (VarExpr (BoolType) b)))))
(ShowCmd
 (BinopExpr
  (BoolType)
  (SumLoopExpr
   (FloatType)
   c
   (IntExpr (IntType) 764)
   d
   (UnopExpr (IntType) - (VarExpr (IntType) argnum))
   (TupleIndexExpr
    (FloatType)
    (TupleLiteralExpr
     (TupleType (IntType) (BoolType) (FloatType))
     (VarExpr (IntType) c)
     (UnopExpr (BoolType) ! (FalseExpr (BoolType)))
     (FloatExpr (FloatType) 23))
    2))
  <
  (FloatExpr (FloatType) 24)))
(TimeCmd
 (LetCmd
  (ArgLValue (ArrayArgument c d e f))
  (TupleIndexExpr
   (ArrayType (IntType) 3)
   (TupleLiteralExpr
    (TupleType (ArrayType (IntType) 3))
    (ArrayLoopExpr
     (ArrayType (IntType) 3)
     c
     (BinopExpr (IntType) (IntExpr (IntType) 141) - (VarExpr (IntType) argnum))
     d
     (IfExpr
      (IntType)
      (VarExpr (BoolType) b)
      (IntExpr (IntType) 12)
      (VarExpr (IntType) argnum))
     e
     (VarExpr (IntType) argnum)
     (IfExpr
      (IntType)
      (FalseExpr (BoolType))
      (VarExpr (IntType) e)
      (IntExpr (IntType) 612))))
   0)))
(LetCmd
 (TupleLValue
  (ArgLValue (VarArgument g))
  (ArgLValue (VarArgument h))
  (TupleLValue (ArgLValue (VarArgument i)) (ArgLValue (VarArgument j))))
 (TupleLiteralExpr
  (TupleType
   (ArrayType (BoolType) 1)
   (BoolType)
   (TupleType (BoolType) (IntType)))
  (ArrayLiteralExpr
   (ArrayType (BoolType) 1)
   (UnopExpr (BoolType) ! (TrueExpr (BoolType))))
  (TupleIndexExpr
   (BoolType)
   (TupleLiteralExpr (TupleType (BoolType)) (TrueExpr (BoolType)))
   0)
  (TupleIndexExpr
   (TupleType (BoolType) (IntType))
   (TupleLiteralExpr
    (TupleType (TupleType (BoolType) (IntType)))
    (TupleLiteralExpr
     (TupleType (BoolType) (IntType))
     (BinopExpr (BoolType) (VarExpr (BoolType) b) || (TrueExpr (BoolType)))
     (SumLoopExpr (IntType) g (VarExpr (IntType) argnum) (VarExpr (IntType) a))))
   0)))
(LetCmd
 (ArgLValue (VarArgument k))
 (ArrayLiteralExpr
  (ArrayType (TupleType (BoolType) (IntType) (IntType)) 1)
  (TupleLiteralExpr
   (TupleType (BoolType) (IntType) (IntType))
   (VarExpr (BoolType) i)
   (VarExpr (IntType) a)
   (SumLoopExpr
    (IntType)
    k
    (TupleIndexExpr
     (IntType)
     (TupleLiteralExpr
      (TupleType (IntType) (IntType))
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) e))
     0)
    l
    (IfExpr
     (IntType)
     (BinopExpr (BoolType) (VarExpr (IntType) a) != (VarExpr (IntType) j))
     (IntExpr (IntType) 522)
     (IntExpr (IntType) 941))
    m
    (VarExpr (IntType) argnum)
    (IntExpr (IntType) 922)))))
(LetCmd (ArgLValue (VarArgument l)) (VarExpr (IntType) argnum))
(LetCmd
 (ArgLValue (VarArgument m))
 (TupleIndexExpr
  (FloatType)
  (TupleIndexExpr
   (TupleType (FloatType) (IntType))
   (TupleLiteralExpr
    (TupleType (TupleType (FloatType) (IntType)) (BoolType) (IntType))
    (IfExpr
     (TupleType (FloatType) (IntType))
     (VarExpr (BoolType) i)
     (TupleLiteralExpr
      (TupleType (FloatType) (IntType))
      (FloatExpr (FloatType) 36)
      (VarExpr (IntType) l))
     (TupleLiteralExpr
      (TupleType (FloatType) (IntType))
      (FloatExpr (FloatType) 0)
      (VarExpr (IntType) argnum)))
    (BinopExpr (BoolType) (VarExpr (IntType) d) != (VarExpr (IntType) a))
    (VarExpr (IntType) l))
   0)
  0))
(ReadCmd "n.png" (VarArgument n))
(ReadCmd "o.png" (ArrayArgument o p q))
